lib/puppet/pops/lookup/lookup_adapter.rb

```diff
@@ -448,7 +448,7 @@ def initialize_env_provider(lookup_invocation)
         ep.config = HieraConfigV5.v4_function_config(env_path, 'environment::data', ep)
         ep
       else
-        raise Puppet::Error.new("Environment '#{environment.name}', cannot find environment_data_provider '#{provider_name}'")
+        raise Puppet::Error.new(_("Environment '%{env}', cannot find environment_data_provider '%{provider}'") % { env: environment.name, provider: provider_name })
```

Comment:
@Magisus environment_data_provider is a keyword, and should be left out of translation